### DISCUSSION TITLE
Meilisearch with JS Hooks?

### DISCUSSION THREAD

**Discussion by KelvinConnor** [pocketbase/pocketbase#5130] [Q&A] [answered]
Does anyone happen to have a reference pb_hooks JavaScript Meilisearch implementation they can share or point me to? Especially interested in how the library, directory, and require need to be configured.

I have extracted the cjs version of Meilisearch from the NPM and placed in /pb_hooks dir and am trying 'const { MeiliSearch } = require('meilisearch')'. Unfortunately I am getting a '_GoError: Invalid module at github.com/dop251/goja_nodejs/require.(*RequireModule).require-fm (native)_' error despite trying several different configuration permutations.

I am fairly ignorant of Node/JavaScript in general so I suspect it might be sometime simple I am missing. 

Thanks!

**Comment by ganigeorgiev** [comment, marked as answer]
> I have extracted the cjs version of Meilisearch from the NPM and placed in /pb_hooks dir and am trying 'const { MeiliSearch } = require('meilisearch')'. 

If you have direct file (and not a node_modules package) then you'll need to require it by specifying its name, aka. ```require(`${__hooks}/yourmeilisearch.cjs`)```.
But even doing this, there is no guarantee that the MeiliSearch lib will work with the JSVM because under the hood it probably relies on some node/browser network native function to send HTTP requests. If you want to send external network requests you'll have to use the [`$http.send`](https://pocketbase.io/docs/js-sending-http-requests/) helper or switch to Go.

**Reply by KelvinConnor** [reply]
Thanks for the reply!

Do you (or anyone) know of any simple external libraries that do work? I just want to validate I am setting it up correctly. 

I dropped lodash.js into the hooks folder and get the same results as example.

**Comment by KelvinConnor** [comment]
For others that might run into this issue... ganigeorgiev's solution works. 

My ignorance of JS was the problem. Apparent a backtick " ` " is required for string templates to work.
